app/assets/src/components/views/PipelineViz.jsx

```diff
@@ -0,0 +1,366 @@
+import React from "react";
+import { DataSet, Network } from "vis";
+import { forEach } from "lodash";
+import PropTypes from "prop-types";
+import ReactPanZoom from "@ajainarayanan/react-pan-zoom";
+
+import cs from "./pipeline_viz.scss";
+
+const START_NODE_ID = -1;
+const END_NODE_ID = -2;
+const STAGE_BG_COLOR = "#f8f8f8";
+const NODE_COLOR = "#eaeaea";
+const EDGE_COLOR = "#999999";
+
+class PipelineViz extends React.Component {
+  constructor(props) {
+    super(props);
+    this.stageResults = this.props.stageResults.stages;
+    this.pipelineVersion = this.props.stageResults.pipeline_version;
+    this.stageGraphsInfo = [];
+
+    if (this.props.admin) {
+      this.stageNames = [
+        "Host Filtering",
+        "GSNAPL/RAPSEARCH alignment",
+        "Post Processing",
+        "Experimental",
+      ];
+    } else {
+      this.stageNames = [
+        "Host Filtering",
+        "GSNAPL/RAPSEARCH alignment",
+        "Post Processing",
+      ];
+    }
+
+    this.state = {
+      stagesOpened: [true, true, true, true],
+      zoom: 1,
+    };
+  }
+
+  componentDidMount() {
+    this.renderGraphs();
+  }
+
+  onMouseWheelZoom(e) {
+    const zoomChange = (e.deltaY < 0 ? 1 : -1) * 0.01;
+    this.setState({ zoom: this.state.zoom + zoomChange });
+  }
+
+  toggleStage(index) {
+    const updatedStagesOpened = [...this.state.stagesOpened];
+    updatedStagesOpened[index] = !updatedStagesOpened[index];
+    this.setState({ stagesOpened: updatedStagesOpened });
+  }
+
+  modifyStepNames() {
+    // Strips 'PipelineStep[Run/Generate]' from front of each step name.
+    // TODO(ezhong): Consider adding 'name' field to dag_json later.
+    forEach(this.stageResults, (stageData, _) => {
```

Comment:
I think it'd be better to move stageResults to the state as well since you're manipulating it directly here